File: remove-gpt-chars.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');
const yargs = require('yargs/yargs')
const { hideBin } = require('yargs/helpers')

function commandLineArgs() {
    return yargs(hideBin(process.argv))
        .scriptName("remove-gpt-chars")
        .usage("$0 [args]")
        .describe("i", "Input file path")
        .describe("e", "Encoding of the input file (default: utf8)")
        .describe("o", "Output file path (if not provided, same as input)")
        .demandOption(["i"])
        .argv
}

const argv = commandLineArgs();
const inputPath = argv.i;
const outputPath = argv.o || inputPath;
const encoding = argv.e || 'utf8';

// Read file
let content;
try {
    content = fs.readFileSync(inputPath, encoding);
} catch (err) {
    console.error(`Error reading file: ${err.message}`);
    process.exit(1);
}

// Remove non-printable and non-ASCII characters
const cleaned = content.replace(/[^\x20-\x7E\n\r\t]/g, '');

// Write result
try {
    fs.writeFileSync(outputPath, cleaned, 'utf8');
    console.log(`Cleaned content written to ${outputPath}`);
} catch (err) {
    console.error(`Error writing file: ${err.message}`);
    process.exit(1);
}

File: remove-gpt-chars.py
#!/usr/bin/env python3

import argparse
import sys
import os

def clean_text(text):
    # Keeps printable ASCII and basic whitespace
    return ''.join(c for c in text if 32 <= ord(c) <= 126 or c in '\n\r\t')

def main():
    parser = argparse.ArgumentParser(description="Remove ChatGPT-style special characters from a plain text file.")
    parser.add_argument("-i", "--input", required=True, help="Input text file path")
    parser.add_argument("-o", "--output", help="Output file path (optional). If not set, overwrites input file.")
    parser.add_argument("-e", "--encoding", default="utf8", help="Encoding to use (default: utf8)")

    args = parser.parse_args()

    input_file = args.input
    output_file = args.output or args.input
    encoding = args.encoding

    if not os.path.isfile(input_file):
        print(f"Error: File not found - {input_file}", file=sys.stderr)
        sys.exit(1)

    try:
        with open(input_file, 'r', encoding=encoding) as f:
            content = f.read()
    except Exception as e:
        print(f"Failed to read input file: {e}", file=sys.stderr)
        sys.exit(1)

    cleaned = clean_text(content)

    try:
        with open(output_file, 'w', encoding=encoding) as f:
            f.write(cleaned)
        print(f"Cleaned text written to {output_file}")
    except Exception as e:
        print(f"Failed to write output file: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()
